temfpy/econometrics.py

```diff
@@ -0,0 +1,286 @@
+
+    y : np.array
+        1d numpy array of shape :math:`n_obs` with the observed choices
+
+    x : np.array
+        2d numpy array of shape :math:`(n_obs, nvar)` including the independent variables.
+
+    cov_structure : str
+                    Available options are 'iid' or 'free'.
+
+    integration_method : str
+                         Either 'mc_integration', 'smooth_mc_integration'
+                         or 'gauss_integration'
+
+    Returns:
+    --------
+        loglike : float
+                  The value of the log-likelihood function evaluated at the given parameters.
+    """
+
+    return _multinomial_probit_loglikeobs(
+        params, y, x, cov_structure, integration_method
+    ).sum()
+
+
+def multinomial_probit(formula, data, cov_structure, integration_method, algorithm):
+    r"""Multinomial probit model.
+
+    .. math::
+        i &= 1, \dots, n \\
+        j &= 1, \dots, m \\
+        \beta_j, X_i &\in \mathbb{R}^{k} \\
+        Y_i^{*j} &= X_i^T \beta_j + \varepsilon_j \\
+        Y_i &= \underset{j}{\mathrm{argmax}} \{Y_i^{*j} \mid j = 1, \dots, m \}
+
+
+    Parameters
+    ----------
+    formula : str
+              A patsy formula comprising the dependent and the independent variables.
+
+    data : pandas.DataFrame
+           A pandas data frame with shape :math:`(n, k+1)`
+           including the dependent variable and the independent variables.
+
+    cov_structure : str
+                    Available options are 'iid' or 'free'.
+
+    integration_method : str
+                         Either 'mc_integration', 'smc_integration'
+                         or 'gauss_integration'
+
+    algorithm : str
+                Available options are 'scipy_lbfgsb' or 'scipy_slsqp'
+
+
+    Returns
+    -------
+    result: dic
+                 Information of the optimization.
+
+
+    Notes
+    -----
+        The function fits a multinomial probit model to discrete choice
+        data via maximum likelihood estimation.
+
+
+    References
+    ----------
+
+    Examples
+    --------
+    >>> import pandas as pd
+    >>> import numpy as np
+    >>> import statsmodels.api as sm
+    >>> import temfpy.econometrics as tpe
+    >>>
+    >>> data = sm.datasets.spector.load_pandas().data
+    >>> f = 'GRADE ~ GPA + TUCE + PSI'
+    >>> np.random.seed(123)
+    >>> data['GRADE'] = np.random.randint(4, size=(32,))
+    >>> cov = 'iid'
+    >>> integr = 'gauss_integration'
+    >>> algo = 'scipy_lbfgsb'
+    >>> solution = tpe.multinomial_probit(f, data, cov, integr , algo)
+    """
+
+    y, x, params = _multinomial_processing(formula, data, cov_structure)
+
+    params_df = pd.DataFrame(params, columns=["value"])
+
+    if cov_structure == "iid":
+        constraints = []
+
+    else:
+        constraints = [
+            {"loc": "covariance", "type": "covariance"},
+            {"loc": ("covariance", 0), "type": "fixed", "value": 1.0},
+        ]

```

Comment:
I'm not sure the docstring for `cov_structure` is accurate. It states it should be either "iid" or "free" but seems like it will accept anything if it's not "iid" (i.e. "free" will return the same as any other input). I would maybe change "free" to None or maybe even raise an exception to help users along if they enter any other value.